### PR TITLE
updated find user endpoint

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/773

### CODE REVIEW

**Review by djino97 — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

File: test/UserService.Business.UnitTests/FindUserCommandTests.cs [hunk 1]
```diff
@@ -130,32 +130,32 @@ public void SetUp()
...
+        //       .Throws(new Exception());
+
+        //    SerializerAssert.AreEqual(result, _command.Execute(_skipCount, _takeCount, _departmentId));
+
+        //    _mocker.Verify<IRequestClient<IFindDepartmentUsersRequest>, Task<Response<IOperationResult<IFindDepartmentUsersResponse>>>>(
+        //        x => x.GetResponse<IOperationResult<IFindDepartmentUsersResponse>>(
+        //            IFindDepartmentUsersRequest.CreateObj(_departmentId, _skipCount, _takeCount), default, It.IsAny<RequestTimeout>()), Times.Once);
+        //    _mocker.Verify<IUserRepository, IEnumerable<DbUser>>(x => x.Get(It.IsAny<List<Guid>>()), Times.Never);
+        //    _mocker.Verify<IUserInfoMapper, UserInfo>(x => x.Map(It.IsAny<DbUser>(), null, null, null, null, null), Times.Never);
+        //}
```
[the remark this answers is not in the document]
Comment (TIove) [reply test/UserService.Business.UnitTests/FindUserCommandTests.cs:158]: Because if it is hard to change tests, we skip it

**Review by TIove — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 2]
```diff
@@ -18,10 +24,99 @@ public class FindUserCommand : IFindUserCommand
...
-        private readonly IRequestClient<IGetUsersDepartmentsUsersPositionsRequest> _getUsersDepartmentsUsersPositionsRequestClient;
+        private readonly IRequestClient<IFindDepartmentUsersRequest> _rcFindDepartmentUser;
+        private readonly IRequestClient<IGetUsersDepartmentsUsersPositionsRequest> _rcGetUsersDepartmentsUsersPositions;
+        private readonly IRequestClient<IGetUserRolesRequest> _rcGetUserRoles;
+        private readonly IRequestClient<IGetUserOfficesRequest> _rcGetUserOffices;
+        private readonly IRequestClient<IGetImagesRequest> _rcGetImages;
+
+        private List<ImageData> GetImages(List<Guid> imageIds, List<string> errors)
+        {
+            string logMessage = "Can not get images.";
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/FindUserCommand.cs:42]: fixed

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 3]
```diff
@@ -18,10 +24,99 @@ public class FindUserCommand : IFindUserCommand
...
+            }
+
+            errors.Add(errorMessage);
+
+            return new();
+        }
+
+        private List<RoleData> GetRoles(List<Guid> userIds, List<string> errors)
+        {
+            string logMessage = "Can not get roles.";
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/FindUserCommand.cs:69]: fixed

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 4]
```diff
@@ -18,10 +24,99 @@ public class FindUserCommand : IFindUserCommand
...
+            }
+
+            errors.Add(errorMessage);
+
+            return new();
+        }
+
+        private List<OfficeData> GetOffice(List<Guid> userIds, List<string> errors)
+        {
+            string logMessage = "Can not get offices.";
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/FindUserCommand.cs:96]: fixed

**Review by djino97 — APPROVED** [review]

**Pull request merged**